qiita_plugins/biom_type/biom_type_plugin/summary.py

```diff
@@ -31,10 +44,79 @@ def generate_html_summary(qclient, job_id, parameters, out_dir):
     ValueError
         If there is any error gathering the information from the server
     """
-    # Only one paramater
-    # artifact_id = parameters['artifact']
-    # TODO
     # Step 1: gather file information from qiita using REST api
+    artifact_id = parameters['input_data']
+    qclient_url = "/qiita_db/artifacts/%s/filepaths/" % artifact_id
+    fps_info = qclient.get(qclient_url)
+    if not fps_info or not fps_info['success']:
+        error_msg = "Could not get artifact filepath information: %s"
+        if fps_info:
+            error_msg = error_msg % fps_info['error']
+        else:
+            error_msg = error_msg % "could not connect with the server"
+        raise ValueError(error_msg)
+
+    # if we get to this point of the code we are sure that this is a biom file
+    # and that it only has one element
+    fps, fps_type = fps_info['filepaths'][0]
+
     # Step 2: generate HTML summary
+    # Modified from https://goo.gl/cUVHgB
+    biom = load_table(fps)
+    num_features, num_samples = biom.shape
+
+    sample_count_summary, sample_counts = count_summary(biom, axis='sample')
+    ax = sns.distplot(sample_counts)
+    ax.set_xlabel("Number of sequences per sample")
+    ax.set_ylabel("Frequency")
+    plot = ax.get_figure()
+    sc_plot = StringIO()
+    plot.savefig(sc_plot, format='png')
+    sc_plot.seek(0)
+
+    uri = 'data:image/png;base64,' + quote(b64encode(sc_plot.buf))
+    artifact_information = [
+        "<b>Number of samples:</b> %d<br/>" % num_samples,
+        "<b>Number of features:</b> %d<br/>" % num_features,
+        ("<b>Minimum count:</b> %d<br/>" %
+         sample_count_summary['Minimum count']),
+        ("<b>Maximum count:</b> %d<br/>" %
+         sample_count_summary['Maximum count']),
+        ("<b>Median count:</b> %d<br/>" %
+         sample_count_summary['Median count']),
+        ("<b>Mean count:</b> %d<br/>" %
+         sample_count_summary['Mean count']),
+        '<br/><hr/><br/>',
+        '<img src = "%s"/>' % uri
+    ]
+
+    of_fp = join(out_dir, "%s.html" % basename(fps))
+    of = open(of_fp, 'w')
+    of.write('\n'.join(artifact_information))
+
     # Step 3: add the new file to the artifact using REST api
-    pass
+    reply = qclient.patch(qclient_url, 'add', '/html_summary/', value=of_fp)
+
+    return reply if not return_html else (reply, artifact_information)
+
+
+# Based on methods from: https://goo.gl/8uebr7
+
+def _counts(table, axis='sample'):
+    result = {}
+    for count_vector, id_, _ in table.iter(axis=axis):
+        result[id_] = float(count_vector.sum())
+    return result
+
+
+def count_summary(table, axis='sample'):
+    counts = _counts(table, axis=axis)
+    counts_values = asarray(counts.values())
+
+    summary = {
+        'Minimum count': min(counts_values),
```

Comment:
if this is already a numpy array, I would use the min method instead of the min function:

```
In [4]: %timeit x = np.random.rand(1000); x.min()
The slowest run took 177.35 times longer than the fastest. This could mean that an intermediate result is being cached 
100000 loops, best of 3: 18.4 µs per loop

In [5]: %timeit x = np.random.rand(1000); min(x)
10000 loops, best of 3: 78.2 µs per loop
```